src/tracer.h

```diff
@@ -174,22 +183,50 @@ class TraceManager {
         std::chrono::duration_cast<std::chrono::nanoseconds>(
             std::chrono::steady_clock::now().time_since_epoch());
 
-    opentelemetry::nostd::shared_ptr<otel_trace_api::Span> trace_span_{nullptr};
-
     std::unique_ptr<otel_trace_sdk::SpanExporter> exporter_;
 
     std::unique_ptr<otel_trace_sdk::SpanProcessor> processor_;
 
     std::shared_ptr<otel_trace_api::TracerProvider> provider_;
 
-    // Initializes Opentelemetry exporter, processor and provider
+    // OTel context to store spans, created in the current trace
+    opentelemetry::context::Context otel_context_;
+
+    // Initializes Opentelemetry exporter, processor, provider and context
     void InitTracer(const TraceConfigMap& config_map);
 
     // Starts a span with the provided timestamp
-    void InitSpan(const opentelemetry::common::SystemTimestamp& timestamp_ns);
-
-    // Ends the started span and cleans up tracer
-    void EndSpan();
+    opentelemetry::nostd::shared_ptr<otel_trace_api::Span> InitSpan(
```

Comment:
`InitSpan -> StartSpan`

Why are we passing two timestamps to this function?

Reply:
One represents the system time to place the span on the timeline, the second is the steady timestamp, for a more precise calculation for span's duration

Reply:
We should add more documentation about these function parameters...